### ISSUE TITLE
Image files are not present in the manifest and not all images are revisioned

### ISSUE THREAD

**Issue by antpaw** [nickjj/manifest-revision-webpack-plugin#12]
Hey, i'm trying to use this plugin, but i fail to make it work the way i need it to:
first, i need all the files form my /images/ dir to be revisioned, even the files I don't require in my js or css. (can this even be done?) and then i'm having this strange bug that the image files are not present in the manifest.json even though they were copied, revisioned and successfully linked.

This is the project with the testcase, just run "npm run webpack" if you want to help me and take a look at it. Thank you very much!
https://github.com/antpaw/webpack-test-project/tree/plugin

**Comment by nickjj** [comment]
Hi, you don't have `webpack` included in your `package.json` so I'm not sure what version you're running.

Can you also try using "webpack": "^1.9.10" and seeing if that changes anything?

**Comment by antpaw** [comment]
sorry, i pushed the fix, it didnt help, exactly the same outcome

**Comment by nickjj** [comment]
Have you tried minimizing it down to just this plugin with your assets, and then add in your other plugins 1 by 1?

**Comment by antpaw** [comment]
okay i have emptied dependencies, file structures, configs in this brach to have the most simplest use case https://github.com/antpaw/webpack-test-project/tree/simple-plugin

and I still have the same two problems

**Comment by nickjj** [comment]
Perfect, thanks. I will look into it over the weekend.

**Comment by nickjj** [comment]
I did glance it quickly and while I haven't tested this yet, why are you going back 2 directories here:
https://github.com/antpaw/webpack-test-project/blob/simple-plugin/webpack.config.js#L22

Also, here's a working example to look at:
https://github.com/nickjj/demo-for-chattanooga-python-user-group

**Comment by antpaw** [comment]
Without the 2 "backs" it crashes because modules cant be found.

the example you gave works, I'm not sure what i'm doing differently, maybe I'm using too many absolute paths. 

another thing: Can you explain when the directory move and revision of rootAssetPath happens? does it happen before anything else in webpack happens?

**Comment by auskast** [comment]
I am having this same issue myself. It seems the problem stems from the if block here https://github.com/nickjj/manifest-revision-webpack-plugin/blob/master/index.js#L60 where it ignores chunked assets. I have a few hundred images in my assets directory that get used statically, but a few of them are imported in my javascript, and none of those imported images make it to the manifest.json, which I still need because they are sometimes accessed in my static html. Maybe we could do with a config option to include certain files even if they appear in chunks?

**Comment by nickjj** [comment]
Is there ever a situation where you'd have a ton of those white listed files? It seems like it would be really tedious if you had to add dozens or hundreds of them to a list.

**Comment by auskast** [comment]
Well the main issue is that 90% of our site is served statically from flask via jinja templates. Almost all of our images are used there, but we want to be able to use webpack and flask-webpack for the 10% that we use react for. I was able to work around it by just never pulling images in through js imports, but obviously that's not ideal as a long-term solution.

**Comment by nickjj** [comment]
Yeah I'm just concerned about a bigger project where you may have all of your images imported in JS. That would mean you'd need to white list all of them, and keeping a manual manifest of all of your assets seems lame.

I wonder if there's a way to do this in a way that's automated.

**Comment by stoikerty** [comment]
I had a similar issue and created a [PR](https://github.com/nickjj/manifest-revision-webpack-plugin/pull/19) that fixed it in my case.
Here is the constructor I'm using:
```js
new ManifestRevisionPlugin(
  path.resolve(PATHS.build, 'manifest.json'),
  {
    rootAssetPath: PATHS.rootAssetPath,
    ignorePaths: [],
    extensionsRegex: /\.(jpe?g|png|gif|svg)$/i,
  }
),
```

@antpaw @auskast 
Could you try out the branch where I have the change and report if it works for you?
To try it out you can use this in your `package.json` (but don't use this in production!)
```json
"manifest-revision-webpack-plugin": "git://github.com/stoikerty/manifest-revision-webpack-plugin.git#feature/additional-regex-option",
```

**Comment by nickjj** [comment]
Definitely interested to see if it works for them, if so that looks like a clean fix.

**Comment by antpaw** [comment]
nope it doesnt work, same problem, you can checkout yourself by installing the git i have included in the intial post

**Comment by antpaw** [comment]
Yes it solved one problem!!! but the images that are not included or required in the code will not appear in the manifest, which is a far less of a problem, but i still need to find a solution for this. I have updated my example git repo as well.

**Comment by stoikerty** [comment]
Nice to hear that it works for you @antpaw!

IIRC images that are not included can't be detected since they are never being traversed. I think that might be out of scope for this plugin. I'm not sure how useful it would be to have a hashed image in your build folder if it's never being used in your code.

What you could do however is simply have a file that references all the images you need. If I add `require('../images/asofoa/aof/imgres.jpg');` to your `foo.js` for instance, the image will appear where you would expect and be referenced in the manifest file.

**Comment by nickjj** [comment]
Thanks for the report.

**Comment by antpaw** [comment]
I need this because I'm building a isomorphic javascript app that also renders on node. So not every image will be present in required frontend js. But i still need one manifest for all my images. However this is no longer an issue for this plugin. Thank you. Will this fix be merged?

**Comment by nickjj** [comment]
@antpaw I merged it yesterday and pushed v0.3 to npm with that fix.

Going to close this for now.
